src/app/functions/server/dashboard.ts

```diff
@@ -0,0 +1,153 @@
+      // data breaches
+      socialSecurityNumbers: 0,
+      ipAddresses: 0,
+      passwords: 0,
+      creditCardNumbers: 0,
+      pinNumbers: 0,
+      securityQuestions: 0,
+    },
+    sanitizedExposures: [],
+  };
+
+  // calculate broker summary from scanned results
+  if (scannedResults) {
+    scannedResults.forEach((r) => {
+      // count email
+      summary.totalExposures += r.emails.length;
+      summary.allExposures.emailAddresses += r.emails.length;
+
+      // count phones
+      summary.totalExposures += r.phones.length;
+      summary.allExposures.phoneNumbers += r.phones.length;
+
+      // count physical addresses
+      summary.totalExposures += r.addresses.length;
+      summary.allExposures.addresses += r.addresses.length;
+
+      // count relatives
+      summary.totalExposures += r.relatives.length;
+      summary.allExposures.familyMembers += r.relatives.length;
+
+      // count full name
+      summary.totalExposures++;
+      summary.allExposures.fullNames++;
+    });
+  }
+
+  // calculate breaches summary from breaches data
+  if (breachesData.verifiedEmails) {
+    for (const emailBreaches of breachesData.verifiedEmails) {
+      const breaches = emailBreaches.breaches;
+      breaches.forEach((b) => {
+        const dataClasses = b.DataClasses;
+
+        // count password
+        if (dataClasses?.includes(BreachDataTypes.Passwords)) {
+          summary.totalExposures++;
+          summary.allExposures.passwords++;
+        }
+
+        // count ssn
+        if (dataClasses?.includes(BreachDataTypes.SSN)) {
+          summary.totalExposures++;
+          summary.allExposures.socialSecurityNumbers++;
+        }
+
+        // count IP
+        if (dataClasses?.includes(BreachDataTypes.IP)) {
+          summary.totalExposures++;
+          summary.allExposures.ipAddresses++;
+        }
+
+        // count credit card
+        if (dataClasses?.includes(BreachDataTypes.CreditCard)) {
+          summary.totalExposures++;
+          summary.allExposures.creditCardNumbers++;
+        }
+
+        // count pin numbers
+        if (dataClasses?.includes(BreachDataTypes.PIN)) {
+          summary.totalExposures++;
+          summary.allExposures.pinNumbers++;
+        }
+
+        // count security questions
+        if (dataClasses?.includes(BreachDataTypes.SecurityQuestions)) {
+          summary.totalExposures++;
+          summary.allExposures.securityQuestions++;
+        }
+      });
+    }
+  }
+
+  return sanitizeExposures(summary);
+}
+
+function sanitizeExposures(summary: DashboardSummary): DashboardSummary {
+  const { allExposures } = summary;
+  const sortedExposures = Object.entries(allExposures).sort(
+    (a, b) => b[1] - a[1]
+  );
+
+  const other =
+    summary.totalExposures -
+    sortedExposures.slice(0, 4).reduce((acc, cur) => acc + cur[1], 0);
+
+  const sanitizedExposures = sortedExposures
+    .map((e) => {
+      const key = e[0];
+      return { [key]: e[1] };
+    })
+    .splice(0, 4);
```

Comment:
Not sure if we should convert the magic numbers `4` here and on L140 into a const.
Q: Also, why `.splice(0, 4)` here but `.slice(0, 4)` on L140 (`splice`-vs-`slice`):

Reply:
Simplified the code a bit, thanks for the suggestion